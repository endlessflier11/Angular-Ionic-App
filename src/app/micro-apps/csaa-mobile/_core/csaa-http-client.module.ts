import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
