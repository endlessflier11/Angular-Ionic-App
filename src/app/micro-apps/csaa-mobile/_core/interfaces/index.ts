export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
