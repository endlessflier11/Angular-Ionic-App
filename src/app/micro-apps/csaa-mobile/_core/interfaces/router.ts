import { Route, UrlTree } from '@angular/router';
import { NavigationOptions } from '@ionic/angular/providers/nav-controller';
import { Observable } from 'rxjs';

interface AppRoute extends Route {
  name?: string;
  isGroupIndex?: boolean; // If true, it means route is the index of parent. Then it inherits all parent properties
}

export type AppRoutes = AppRoute[];

export interface RouteParam {
  [key: string]: string;
}

export interface NamedRouteConfig {
  name: string;
  path: string;
  children?: NamedRouteConfig[];
}

export interface NamedRoute {
  name: string;
  path: string;
  relativePath: string;
}

export interface CanComponentDeactivate {
  canDeactivate: () =>
    | Observable<boolean | UrlTree>
    | Promise<boolean | UrlTree>
    | boolean
    | UrlTree;
}

export interface CsaaRouteStackItem {
  name: string;
  params?: RouteParam;
  options?: NavigationOptions;
}
