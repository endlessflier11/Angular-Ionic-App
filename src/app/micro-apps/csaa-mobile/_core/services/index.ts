export * from './router/router.service';
export * from './config/config.service';
export * from './auth/auth.service';
export * from './sso/sso.service';
export * from './storage/storage.service';
export * from './http/http.service';
export * from './analytics.service';
export * from './policy.service';
export * from './payment.service';
export * from './global-state.service';
export * from './call.service';
export * from './metadata.service';
export * from './pdf-display.service';
export * from './webview/webview.service';
