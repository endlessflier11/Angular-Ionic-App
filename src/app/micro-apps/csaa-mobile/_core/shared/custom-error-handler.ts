import { ErrorHandler, Injectable } from '@angular/core';
import { AlertController } from '@ionic/angular';

import { CsaaCoreModule } from '../../csaa-core/csaa-core.module';

export class RefreshTokenExpiredError extends Error {
  constructor(...params) {
    super(...params);
    // Set the prototype explicitly.
    Object.setPrototypeOf(this, RefreshTokenExpiredError.prototype);
  }
}
export class UnknownRefreshError extends Error {
  constructor(...params) {
    super(...params);
    // Set the prototype explicitly.
    Object.setPrototypeOf(this, UnknownRefreshError.prototype);
  }
}

@Injectable({
  providedIn: CsaaCoreModule,
})
export class CustomErrorHandler implements ErrorHandler {
  constructor(
    private alertCtrl: AlertController // TODO v5: keep?  private signOut: SignOutDirective
  ) {}

  handleError(error: any) {
    const httpErrorCode = error.httpErrorCode;
    console.error(error);
    switch (httpErrorCode) {
      // todo: ... do we care to give a different message for each error?
      default:
        this.showError(error);
    }
  }

  async getAlertForError(error) {
    if (error instanceof RefreshTokenExpiredError) {
      return await this.alertCtrl.create({
        header: `Login expired`,
        subHeader: `Please sign up again.`,
        buttons: [
          {
            text: 'Okay',
            role: 'cancel',
            handler: () => 
              // TODO v5: keep?
              // this.signOut.signOut();
               true
            ,
          },
        ],
      });
    }

    if (error instanceof UnknownRefreshError) {
      return this.alertCtrl.create({
        header: 'Something went wrong',
        subHeader: `We're sorry, there was an issue. Tap Dismiss then pull-down to refresh screen or try again later.`,
        buttons: [
          {
            text: 'Dismiss',
            role: 'cancel',
          },
        ],
      });
    }
    if (error.status && error.body && error.body.error) {
      return this.alertCtrl.create({
        header: `${error.status} error`,
        subHeader: error.body.error,
        buttons: [
          {
            text: 'Dismiss',
            role: 'cancel',
          },
        ],
      });
    }
    return this.alertCtrl.create({
      header: 'An error occurred',
      subHeader: 'Retry or contact support',
      message: error.message,
      buttons: [
        {
          text: 'Dismiss',
          role: 'cancel',
        },
      ],
    });
  }

  private async showError(error) {
    const alert = await this.getAlertForError(error);
    await alert.present();
  }
}
