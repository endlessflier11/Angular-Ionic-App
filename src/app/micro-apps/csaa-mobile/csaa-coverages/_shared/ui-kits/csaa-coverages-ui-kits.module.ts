import { NgModule } from '@angular/core';
import { CsaaPupCoveragesCardComponent } from './csaa-pup-coverages-card/csaa-pup-coverages-card.component';
import { CsaaGeneralCoveragesCardComponent } from './csaa-general-coverages-card/csaa-general-coverages-card.component';
import { CsaaDriversCoveragesCardComponent } from './csaa-drivers-coverages-card/csaa-drivers-coverages-card.component';
import { CsaaHomeCoveragesCardComponent } from './csaa-home-coverages-card/csaa-home-coverages-card.component';
import { CsaaVehiclesCoveragesCardComponent } from './csaa-vehicles-coverages-card/csaa-vehicles-coverages-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaCoverageCardItemComponent } from './csaa-coverage-card-item/csaa-coverage-card-item.component';
import {
  CsaaCoveragesVehicleCoveredCardComponent
} from './csaa-vehicle-coverages-covered-card/csaa-coverages-vehicle-covered-card.component';
import {
  CsaaCoveragesVehicleNotCoveredCardComponent
} from './csaa-vehicle-coverages-not-covered-card/csaa-coverages-vehicle-not-covered-card.component';
import { CsaaDriversCoveragesContentV1Component } from './csaa-drivers-coverages-content-v1/csaa-drivers-coverages-content-v1.component';
import { CsaaDriversCoveragesContentV2Component } from './csaa-drivers-coverages-content-v2/csaa-drivers-coverages-content-v2.component';

const SHARED_COMPONENTS = [
  CsaaCoverageCardItemComponent,
  CsaaPupCoveragesCardComponent,
  CsaaGeneralCoveragesCardComponent,
  CsaaDriversCoveragesCardComponent,
  CsaaDriversCoveragesContentV1Component,
  CsaaHomeCoveragesCardComponent,
  CsaaVehiclesCoveragesCardComponent,
  CsaaCoveragesVehicleCoveredCardComponent,
  CsaaCoveragesVehicleNotCoveredCardComponent,
  CsaaDriversCoveragesContentV2Component,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaCoveragesUiKitsModule {}
