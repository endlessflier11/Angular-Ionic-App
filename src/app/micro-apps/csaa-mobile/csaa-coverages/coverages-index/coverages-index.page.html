<ion-header [ngClass]="['csaa', currentTheme]">
  <ion-toolbar>
    <ion-buttons slot="start">
      <csaa-back-button (back)="backButtonClick()"></csaa-back-button>
    </ion-buttons>
    <ion-buttons slot="end">
      <ion-button (click)="hoursPopup.show()">Call Service</ion-button>
    </ion-buttons>
  </ion-toolbar>
  <ion-toolbar>
    <ion-title>Coverages</ion-title>
  </ion-toolbar>
</ion-header>
<ion-content [ngClass]="['csaa', currentTheme]">
  <ion-refresher (ionRefresh)="doRefresh($event)" slot="fixed">
    <ion-refresher-content></ion-refresher-content>
  </ion-refresher>
  <ng-container *ngIf="policyType === PolicyTypes.Auto">
    <ng-container *ngIf="!allVehiclesWaivedLiability()">
      <csaa-general-coverages-card
        *ngIf="loading || (policy$ | async)?.coverages?.length > 0"
        [policyNumber]="policyNumber"
        [showEditPolicy]="isPbeEligible$ | async"
        [policyType]="policyType"
        [agent]="(policy$ | async)?.agent"
        [loading]="loading"
        [coverages]="(policy$ | async)?.coverages"
        [policy]="policy$ | async"
        (viewDeclarationClick)="viewDeclaration()"
        (webviewDismissed)="doRefresh()"
      >
      </csaa-general-coverages-card>
    </ng-container>
    <csaa-vehicles-coverages-card
      [loading]="loading"
      [policy]="policy$ | async"
      (viewDeclarationClick)="viewDeclaration()"
      [showDeclarations]="(policy$ | async)?.coverages?.length < 1"
    >
    </csaa-vehicles-coverages-card>
    <csaa-drivers-coverages-card [loading]="loading" [drivers]="(policy$ | async)?.drivers">
    </csaa-drivers-coverages-card>
  </ng-container>
  <ng-container *ngIf="policyType === PolicyTypes.PUP">
    <csaa-pup-coverages-card
      [policyNumber]="policyNumber"
      [policy]="policy$ | async"
      [loading]="loading"
      [coverages]="(policy$ | async)?.coverages"
      (viewDeclarationClick)="viewDeclaration()"
    >
    </csaa-pup-coverages-card>
  </ng-container>

  <ng-container *ngIf="policyType === PolicyTypes.Home">
    <csaa-home-coverages-card
      [policyNumber]="policyNumber"
      [loading]="loading"
      [policy]="policy$ | async"
      [coverages]="(policy$ | async)?.coverages"
      (viewDeclarationClick)="viewDeclaration()"
    >
    </csaa-home-coverages-card>
  </ng-container>
</ion-content>
<csaa-working-hours
  #hoursPopup
  [ngClass]="['csaa', currentTheme]"
  [riskState]="(policy$ | async)?.riskState"
  serviceType="CUSTOMER"
>
</csaa-working-hours>
