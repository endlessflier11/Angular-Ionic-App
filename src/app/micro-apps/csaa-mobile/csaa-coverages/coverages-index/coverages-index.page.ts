import { Component, OnDestroy, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { IonRefresher, Platform } from '@ionic/angular';
import { Store } from '@ngxs/store';
import { noop, Observable, Subscription } from 'rxjs';
import { filter, finalize, map, switchMap } from 'rxjs/operators';
import { withErrorReporter } from '../../_core/helpers';
import { unsubscribeIfPresent } from '../../_core/helpers/async.helper';
import { Agent } from '../../_core/interfaces/agent.interface';
import { Category, EventName, EventType } from '../../_core/interfaces/analytics.interface';
import { DocumentType } from '../../_core/interfaces/document.interface';
import { Policy, PolicyType } from '../../_core/interfaces/policy.interface';
import { AnalyticsService } from '../../_core/services/analytics.service';
import { RouterService } from '../../_core/services/router/router.service';
import { CustomErrorHandler } from '../../_core/shared/custom-error-handler';
import { PolicyDocumentHelper } from '../../_core/shared/policy-document.helper';
import { PolicyHelper } from '../../_core/shared/policy.helper';
import { ConfigState } from '../../_core/store/states/config.state';
import { SubSink } from '../../_core/shared/subscription.helper';
import { PolicyState } from '../../_core/store/states/policy.state';
import { CustomerAction, PolicyAction } from '../../_core/store/actions';

@Component({
  selector: 'csaa-coverages-index',
  templateUrl: './coverages-index.page.html',
  styleUrls: ['./coverages-index.page.scss'],
})
export class CsaaCoveragesIndexPage implements OnInit, OnDestroy {
  policyNumber: string;
  policyType: PolicyType;

  currentTheme: string;

  // eslint-disable-next-line @typescript-eslint/naming-convention
  PolicyTypes = PolicyType;
  loading = false;
  vehiclesWithLiabilityCoverage = 0;

  policy$: Observable<Policy>;
  isPbeEligible$: Observable<boolean>;

  agent: Agent;
  isAndroidPlatform: boolean;

  backButtonSubscription: Subscription;

  private refresher: IonRefresher;
  private readonly subsink = new SubSink();

  constructor(
    private store: Store,
    private errorHandler: CustomErrorHandler,
    private analyticsService: AnalyticsService,
    private policyDocumentHelper: PolicyDocumentHelper,
    private platform: Platform,
    private route: ActivatedRoute,
    private routerService: RouterService
  ) {}

  ngOnInit() {
    this.loading = true;
    this.isAndroidPlatform = this.platform.is('android');
    this.currentTheme = this.store.selectSnapshot(ConfigState.theme);
    this.route.paramMap.subscribe(
      withErrorReporter((paramMap) => {
        if (!paramMap.has('policyNumber')) {
          this.routerService.back();
        }

        this.policyNumber = paramMap.get('policyNumber');
        this.policy$ = this.store
          .select(PolicyState.policyData(this.policyNumber))
          .pipe(filter((v) => !!v));

        this.isPbeEligible$ = this.store
          .select(PolicyState.allowedEndorsements(this.policyNumber))
          .pipe(map((allowed) => !!allowed));

        this.subsink.add(
          this.policy$.subscribe(
            withErrorReporter(
              (policy) => {
                this.policyType = policy.type;
                this.agent = policy.agent;
                this.vehiclesWithLiabilityCoverage = policy.vehicles.length;
                policy.vehicles.forEach((vehicle) => {
                  if (vehicle.riskFactors.waivedLiability === true) {
                    this.vehiclesWithLiabilityCoverage -= 1;
                  }
                });
              },
              (error) => {
                error.report();
                this.errorHandler.handleError(error);
              }
            )
          )
        );
        this.dispatchLoadAction();
      })
    );
  }

  ngOnDestroy() {
    this.subsink.unsubscribe();
  }

  ionViewWillEnter() {
    this.backButtonSubscription = this.platform.backButton.subscribeWithPriority(10, () =>
      this.backButtonClick()
    );
  }

  ionViewWillLeave() {
    unsubscribeIfPresent(this.backButtonSubscription);
  }

  doRefresh(event = null) {
    this.loading = true;
    this.refresher = event?.target;
    this.dispatchReloadActions();
  }

  viewDeclaration() {
    this.analyticsService.trackEvent(EventName.VIEW_DECLARATIONS, Category.coverages, {
      event_type: EventType.FILE_DOWNLOADED,
      file: 'Declaration',
    });
    const productType = PolicyHelper.prodTypeFromPolicyType(this.policyType);
    this.policyDocumentHelper.openDocument(
      this.policyNumber,
      productType,
      DocumentType.Declarations
    );
  }

  backButtonClick() {
    this.analyticsService.trackEvent(EventName.HOME_ACCESSED, Category.global, {
      event_type: EventType.LINK_ACCESSED,
      link: 'Home',
    });
    this.routerService.back();
  }

  allVehiclesWaivedLiability(): boolean {
    return this.vehiclesWithLiabilityCoverage > 0 ? false : true;
  }

  onAgentContact(method) {
    this.analyticsService.trackEvent(EventName.CONTACT_INITIATED, Category.coverages, {
      event_type: EventType.OPTION_SELECTED,
      selection: 'Call Service',
      method,
    });
  }

  private dispatchLoadAction() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.LoadPolicies())),
        switchMap(() =>
          this.store.dispatch(new PolicyAction.LoadAllowedEndorsements(this.policyNumber))
        ),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter());
  }

  dispatchReloadActions() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.ReloadPolicies())),
        switchMap(() =>
          this.store.dispatch(new PolicyAction.ReloadAllowedEndorsements(this.policyNumber))
        ),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter());
  }

  private completeLoading() {
    this.loading = false;
    if (this.refresher) {
      this.refresher.complete().then(noop);
    }
  }
}
