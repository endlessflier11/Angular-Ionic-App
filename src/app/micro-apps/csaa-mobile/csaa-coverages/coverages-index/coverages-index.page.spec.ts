import { ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
import { IonicModule, NavController } from '@ionic/angular';
import { of } from 'rxjs';
import { CallService } from '../../_core/services/call.service';
import { CustomErrorHandler } from '../../_core/shared/custom-error-handler';
import { PolicyDocumentHelper } from '../../_core/shared/policy-document.helper';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { CsaaVehicleCoveragesPage } from '../vehicle-coverages-page/vehicle-coverages.page';

import { CsaaCoveragesIndexPage } from './coverages-index.page';
import { CsaaCoveragesUiKitsModule } from '../_shared/ui-kits/csaa-coverages-ui-kits.module';
import { ActivatedRoute } from '@angular/router';

import { GlobalStateService } from '../../_core/services';
import { By, clickEvent, PageTestingModule, StoreTestBuilder } from '../../../../../testing';
import { Category, EventName } from '../../_core/interfaces';
import { AnalyticsService } from '../../_core/services/analytics.service';
import { NgxsModule, Store } from '@ngxs/store';
import { CsaaAppInjector } from '../../csaa-app.injector';
import { Injector } from '@angular/core';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { AppState } from '../../../../_core/store';
import { CsaaCoreModule } from '../../csaa-core/csaa-core.module';

describe('CsaaCoveragesIndexPage', () => {
  let component: CsaaCoveragesIndexPage;
  let fixture: ComponentFixture<CsaaCoveragesIndexPage>;
  let analyticService;

  beforeEach(async () => {
    const PARAM_MAP = new Map();
    PARAM_MAP.set('policyNumber', 'CAAC910026006'); // !important - must be one of the polices in the PolicyState
    PARAM_MAP.set('policyType', '2');
    await TestBed.configureTestingModule({
      declarations: [CsaaCoveragesIndexPage, CsaaVehicleCoveragesPage],
      imports: [
        IonicModule,
        UiKitsModule,
        CsaaCoveragesUiKitsModule,
        PageTestingModule.withConfig({
          providesStorage: true,
          providesHttp: true,
          providesPlatform: true,
          providesRouter: true,
          providesConfig: true,
          providesAnalytics: true,
        }),
        HttpClientTestingModule,
        NgxsModule.forRoot([AppState]),
        CsaaCoreModule.forRoot(),
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            paramMap: of(PARAM_MAP),
          },
        },
        { provide: NavController, useValue: { push: jest.fn() } },
        {
          provide: CallService,
          useValue: {
            call: jest.fn(),
            getServiceNumber: jest.fn(() => '12345'),
            getEmergencyNumber: jest.fn(() => '45678'),
            getClaimsNumber: jest.fn(() => '1234'),
          },
        },
        { provide: CustomErrorHandler, useValue: { handleError: jest.fn() } },
        {
          provide: PolicyDocumentHelper,
          useValue: {
            openDocument: jest.fn(),
          },
        },
        {
          provide: GlobalStateService,
          useValue: {
            getIsStandalone: jest.fn(() => false),
            getRegistrationId: jest.fn(),
          },
        },
      ],
    }).compileComponents();

    const store = TestBed.inject(Store);
    StoreTestBuilder.withDefaultMocks().resetStateOf(store);

    CsaaAppInjector.injector = TestBed.inject(Injector);
    analyticService = TestBed.inject(AnalyticsService);
    fixture = TestBed.createComponent(CsaaCoveragesIndexPage);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    component.ngOnInit();
    component.ionViewWillEnter();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should track the view declarations button click', fakeAsync(() => {
    component.ngOnInit();
    component.ionViewWillEnter();

    component.viewDeclaration();
    expect(analyticService.trackEvent).toHaveBeenCalledWith(
      EventName.VIEW_DECLARATIONS,
      Category.coverages,
      {
        event_type: 'File Downloaded',
        file: 'Declaration',
      }
    );
  }));

  it('should track call service clicked', () => {
    const callBtn = fixture.debugElement.query(
      By.cssAndText('csaa-working-hours ion-button', 'Call Service')
    );
    callBtn.triggerEventHandler('click', clickEvent);
    expect(analyticService.trackEvent).toHaveBeenCalledWith(
      EventName.CONTACT_INITIATED,
      Category.global,
      {
        event_type: 'Option Selected',
        method: 'phone',
        selection: 'Call Service',
      }
    );
  });

  it('should track Home Accessed after  back button click', () => {
    component.backButtonClick();
    expect(analyticService.trackEvent).toHaveBeenCalledWith(
      EventName.HOME_ACCESSED,
      Category.global,
      {
        event_type: 'Link Accessed',
        link: 'Home',
      }
    );
    expect(analyticService.trackEvent).toHaveBeenCalledTimes(1);
  });

  it('should track agent contact', () => {
    component.onAgentContact('phone');
    expect(analyticService.trackEvent).toHaveBeenCalledWith(
      EventName.CONTACT_INITIATED,
      Category.coverages,
      {
        event_type: 'Option Selected',
        method: 'phone',
        selection: 'Call Service',
      }
    );
  });
});
