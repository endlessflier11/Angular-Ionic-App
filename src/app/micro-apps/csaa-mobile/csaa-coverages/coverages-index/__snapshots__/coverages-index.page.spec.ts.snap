// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaCoveragesIndexPage should match snapshot 1`] = `
<csaa-coverages-index
  PolicyTypes={[Function Object]}
  __ngContext__={[Function LRootView]}
  agent={[Function Object]}
  analyticsService={[Function AnalyticsMockService]}
  backButtonSubscription={[Function Observable]}
  currentTheme=""
  errorHandler={[Function Object]}
  isAndroidPlatform="false"
  isPbeEligible$={[Function Observable]}
  loading="false"
  platform={[Function PlatformMock]}
  policy$={[Function Observable]}
  policyDocumentHelper={[Function Object]}
  policyNumber={[Function String]}
  policyType={[Function Number]}
  route={[Function Object]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  subsink={[Function SubSink]}
  vehiclesWithLiabilityCoverage={[Function Number]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Service
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Coverages
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <csaa-general-coverages-card>
      <ion-card
        class="csaa-card coverage-list"
      >
        <ion-card-header>
          <ion-item>
            <ion-card-title>
              General
            </ion-card-title>
          </ion-item>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
            <ion-item>
              <csaa-coverage-card-item
                class="no-border"
              >
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Bodily Injury
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                      <span>
                        $100,000
                      </span>
                       Per Person 
                      <span>
                        / $300,000
                      </span>
                       Per Accident 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Property Damage
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                      <span>
                        $50,000
                      </span>
                       Per Accident 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Uninsured Motorist
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                      <span>
                        $100,000
                      </span>
                       Per Person 
                      <span>
                        / $300,000
                      </span>
                       Per Accident 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Medical Payments
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                      <span>
                        $2,000
                      </span>
                       Per Person 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
            View Declarations
          </ion-button>
        </ion-card-content>
      </ion-card>
    </csaa-general-coverages-card>
    <csaa-vehicles-coverages-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
            Vehicle
          </ion-card-title>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
            <ion-item
              detail=""
            >
              <h3>
                 2009 Acura Tsx 
              </h3>
            </ion-item>
            <ion-item
              detail=""
            >
              <h3>
                 2007 Jeep Grand Cherokee 
              </h3>
            </ion-item>
            <ion-item
              detail=""
            >
              <h3>
                 2015 Kia Forte 
              </h3>
            </ion-item>
            <ion-item
              detail=""
            >
              <h3>
                 2006 Ford Lgt Convtnl F 
              </h3>
            </ion-item>
          </ion-list>
        </ion-card-content>
      </ion-card>
    </csaa-vehicles-coverages-card>
    <csaa-drivers-coverages-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
            Drivers
          </ion-card-title>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <csaa-csaa-drivers-coverages-content-v1>
            <ion-list>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Haithm Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                       No Extra Coverages 
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Neesa Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                       No Extra Coverages 
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Nana Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                       No Extra Coverages 
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Sana Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                      Excluded Driver
                    </p>
                  </div>
                </ion-label>
              </ion-item>
              <ion-item>
                <ion-label>
                  <h3
                    class="driver-name"
                  >
                    Lola Alabdi
                  </h3>
                  <div
                    class="description"
                  >
                    <p>
                       No Extra Coverages 
                    </p>
                  </div>
                </ion-label>
              </ion-item>
            </ion-list>
          </csaa-csaa-drivers-coverages-content-v1>
        </ion-card-content>
      </ion-card>
    </csaa-drivers-coverages-card>
  </ion-content><csaa-working-hours
    class="csaa"
    servicetype="CUSTOMER"
  >
    <div
      class="csaa-working-hours hidden"
    >
      <ion-card
        class="card card-md"
      >
        <ion-card-header>
          Service Hours
        </ion-card-header>
        <ion-card-content
          class="card-content card-content-md"
        >
          <ion-list>
            <ion-item>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
             Call Service 
          </ion-button>
        </ion-card-content>
      </ion-card>
    </div>
  </csaa-working-hours>
</csaa-coverages-index>
`;
